src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.CodeModelEventCollector.cs

```diff
@@ -970,29 +973,38 @@ protected override void EnqueueRemoveEvent(SyntaxNode node, SyntaxNode parent, C
                 {
                     foreach (var variableDeclarator in ((BaseFieldDeclarationSyntax)node).Declaration.Variables)
                     {
-                        eventQueue.EnqueueRemoveEvent(variableDeclarator, parent);
+                        eventQueue.EnqueueChangeEvent(variableDeclarator, parent, eventType);
                     }
                 }
                 else if (node is AttributeListSyntax)
                 {
                     foreach (var attribute in ((AttributeListSyntax)node).Attributes)
                     {
-                        if (parent is BaseFieldDeclarationSyntax)
-                        {
-                            foreach (var variableDeclarator in ((BaseFieldDeclarationSyntax)parent).Declaration.Variables)
-                            {
-                                eventQueue.EnqueueRemoveEvent(attribute, variableDeclarator);
-                            }
-                        }
-                        else
-                        {
-                            eventQueue.EnqueueRemoveEvent(attribute, parent);
-                        }
+                        ChangeEventQueueForAttributes(attribute, parent, eventType, eventQueue);
                     }
                 }
+                else if (node is AttributeSyntax)
+                {
+                    ChangeEventQueueForAttributes((AttributeSyntax)node, parent, eventType, eventQueue);
+                }
                 else
                 {
-                    eventQueue.EnqueueRemoveEvent(node, parent);
+                    eventQueue.EnqueueChangeEvent(node, parent, eventType);
+                }
+            }
+
+            private static void ChangeEventQueueForAttributes(AttributeSyntax attribute, SyntaxNode parent, CodeModelEventType eventType, CodeModelEventQueue eventQueue)
```

Comment:
Isn't this identical to AddOrRemoveEventQueueForAttribute but using eventType instead of the lambda?